solidity/contracts/converter/types/liquidity-pool-v2/LiquidityPoolV2Converter.sol

```diff
@@ -1019,101 +700,62 @@ contract LiquidityPoolV2Converter is LiquidityPoolConverter {
       * @return amplified balance
     */
     function amplifiedBalance(IERC20Token _reserveToken) internal view returns (uint256) {
-        return stakedBalances[_reserveToken].mul(amplificationFactor - 1).add(reserves[_reserveToken].balance);
+        return stakedBalances[_reserveToken].mul(AMPLIFICATION_FACTOR - 1).add(reserves[_reserveToken].balance);
     }
 
     /**
-      * @dev returns the effective primary reserve weight based on the staked balance, current balance and given rate
+      * @dev returns the effective reserve weights based on the staked balance, current balance and oracle price
       *
-      * @param _rate    rate of 1 primary token in secondary tokens
+      * @param _rate    rate between the reserve tokens
       *
-      * @return primary reserve weight
+      * @return new primary reserve weight
+      * @return new secondary reserve weight
     */
-    function primaryWeightFromRate(Fraction memory _rate) private view returns (uint32) {
-        uint256 a = stakedBalances[primaryReserveToken].mul(_rate.n);
-        uint256 b = stakedBalances[secondaryReserveToken].mul(_rate.d);
-        (uint256 x,) = normalizedRatio(a, b, PPM_RESOLUTION);
-        return uint32(x);
-    }
+    function effectiveReserveWeights(Fraction memory _rate) private view returns (uint32, uint32) {
+        // get the primary reserve staked balance
+        uint256 primaryStakedBalance = stakedBalances[primaryReserveToken];
 
-    /**
-      * @dev returns the effective rate based on the staked balance, current balance and given primary reserve weight
-      *
-      * @param _primaryReserveWeight    primary reserve weight
-      *
-      * @return effective rate of 1 primary token in secondary tokens
-    */
-    function rateFromPrimaryWeight(uint32 _primaryReserveWeight) private view returns (Fraction memory) {
-        uint256 n = stakedBalances[secondaryReserveToken].mul(_primaryReserveWeight);
-        uint256 d = stakedBalances[primaryReserveToken].mul(inverseWeight(_primaryReserveWeight));
-        (n, d) = reducedRatio(n, d, MAX_RATE_FACTOR_LOWER_BOUND);
-        return Fraction(n, d);
+        // get the tokens amplified balances
+        uint256 primaryBalance = reserveAmplifiedBalance(primaryReserveToken);

```

Comment:
pretty sure we can call `amplifiedBalance` here to skip the `validReserve` check in `reserveAmplifiedBalance`

Reply:
Fixed in all places; please verify.